### DISCUSSION TITLE
`page.canonical_url` works with local server, but not with build?

### DISCUSSION THREAD

**Discussion by fralau** [squidfunk/mkdocs-material#2289] [Q&A] [answered]
I used [Custom Meta Tags](https://squidfunk.github.io/mkdocs-material/reference/meta-tags/#custom-meta-tags), as recommended, but I am left with a little mystery:

```
{% block extrahead %}
  {% set title = config.site_name %}
  {% if page and page.meta and page.meta.title %}
    {% set title = title ~ " - " ~ page.meta.title %}
  {% elif page and page.title and not page.is_homepage %}
    {% set title = title ~ " - " ~ page.title | striptags %}
  {% endif %}
  <meta property="og:type" content="website" />
  <meta property="og:title" content="{{ title }}" />
  <meta property="og:description" content="{{ config.site_description }}" />
  <meta property="og:url" content="{{ page.canonical_url }}" />
{% endblock %}
```

The meta property `og:url` is correctly filled in the html page, when I do `mkdocs serve`; but in case of build (and publication on the Apache server), it returns None.

Is there some issue with `page.canonical_url` ? Or am I missing something important?

**Comment by squidfunk** [comment, marked as answer]
Did you set [`site_url`](https://www.mkdocs.org/user-guide/configuration/#site_url) in `mkdocs.yml`? Without a `site_url`, [`page.canonical_url`](https://www.mkdocs.org/user-guide/custom-themes/#pagecanonical_url) cannot be deduced.

**Reply by fralau** [reply]
🤦  Obvious...

Thanks a lot for the hint. Problem solved.
